server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -42,14 +44,40 @@ func NewBuildBuddyServer(env environment.Env, sslService *ssl.SSLService) (*Buil
 	}, nil
 }
 
-func (s *BuildBuddyServer) redactAPIKey(ctx context.Context, rsp *inpb.GetInvocationResponse) error {
-	apiKey := s.getGroupAPIKey(ctx)
-	if apiKey == "" {
-		return nil
+func (s *BuildBuddyServer) getConfiguredAPIKey() string {
+	if apiConfig := s.env.GetConfigurator().GetAPIConfig(); apiConfig != nil {
+		return apiConfig.APIKey
 	}
+	return ""
+}
+
+func (s *BuildBuddyServer) redactAPIKeys(ctx context.Context, rsp *inpb.GetInvocationResponse) error {
 	proto.DiscardUnknown(rsp)
 	txt := proto.MarshalTextString(rsp)
-	txt = strings.ReplaceAll(txt, apiKey, "<REDACTED>")
+
+	pat := regexp.MustCompile("x-buildbuddy-api-key=[A-Za-z0-9]{20}")
+	txt = pat.ReplaceAllLiteralString(txt, "x-buildbuddy-api-key=<REDACTED>")
+
+	pat = regexp.MustCompile(fmt.Sprintf(
```

Comment:
FYI I'm testing this logic; so far it looks a bit buggy. Will update this thread once it's fixed

Reply:
Fixed: https://play.golang.org/p/K_aODmaDLr-